internal/utils/utils.go

```diff
@@ -223,3 +226,99 @@ func ReadWithPages[T any](read func(offset, limit int32) ([]T, error)) ([]T, err
 
 	return allResults, nil
 }
+
```

Comment:
We should have UTs defined for this file where possible but this can be done in another PR